src/MAT_v5.jl

```diff
@@ -31,6 +31,14 @@ using Compat.String
 import Base: read, write, close
 import HDF5: names, exists
 
+if VERSION < v"0.6.0-dev.1632"
+    round_uint8(data) = round(UInt8, data)
+    complex_array(a, b) = complex(a, b)
+else
+    round_uint8(data) = round.(UInt8, data)
+    complex_array(a, b) = complex.(a, b)
```

Comment:
And just a note for future reviewers. These are to workaround non-numbers not treated as scalars for broadcast on 0.5 and the bad type inference on 0.4. The version check is not a exact version for any of these to work but should be new enough for these to work and old enough for these to not raise a depwarn.